brave/brave-core
# Add Leo context menu metrics

Resolves https://github.com/brave/brave-browser/issues/36560



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For `Brave.AIChat.ContextMenu.FreeUsages` and `Brave.AIChat.ContextMenu.PremiumUsages`:

0. Start with fresh profile
1. Select some text on a page, and use a Leo context menu option
2. Before opting in, ensure metric does not exist in local state
3. Opt in to Leo. Ensure the free usages metric exists with the correct value.
4. Use the context menu options multiple times, ensure the metric count updates
5. Activate premium. Use the context menu again. Ensure the premium metric exists with the correct value, and the free metric is gone.

For `Brave.AIChat.ContextMenu.MostUsedAction`:

0. Start with fresh profile
1. Select some text on a page, and use a Leo context menu option
2. Before opting in, ensure metric does not exist in local state
3. Opt in to Leo. Ensure the metric exists with the correct value.
4. Use various context menu options, and ensure the metric value updates accordingly.

For `Brave.AIChat.ContextMenu.LastUsageTime`:

0. Start with fresh profile
1. Select some text on a page, and use a Leo context menu option
2. Before opting in, ensure metric does not exist in local state
3. Opt in to Leo. Ensure the metric exists with the value `1`.
4. Close browser, advance system time by a couple weeks, open browser, ensure metric value is correct.
5. Use context menu again, ensure value updates back to `1`.
